pkg/controller/clusterpool/collections.go

```diff
@@ -584,6 +599,252 @@ func (cds *cdCollection) Delete(c client.Client, cdName string) error {
 	return nil
 }
 
+type cdcCollection struct {
+	// Unclaimed by any cluster pool CD any not broken
+	unassigned []*hivev1.ClusterDeploymentCustomization
+	// Listed in the cluster pool inventory but not found
+	missing []string
+	// Used by some cluster deployment
+	reserved []*hivev1.ClusterDeploymentCustomization
+	// Last Cluster Deployment failed on provision
+	cloud []*hivev1.ClusterDeploymentCustomization
+	// Failed to apply patches for this cluster pool
+	syntax []*hivev1.ClusterDeploymentCustomization
+	// All CDCs in this pool
+	byCDCName map[string]*hivev1.ClusterDeploymentCustomization
+}
+
+// getAllCustomizationsForPool is the constructor for a cdcCollection for all of the
+// ClusterDeploymentCustomizations that are related to specified pool.
+func getAllCustomizationsForPool(c client.Client, pool *hivev1.ClusterPool, logger log.FieldLogger) (*cdcCollection, error) {
+	if pool.Spec.Inventory == nil {
+		return nil, nil
+	}
```

Comment:
The other `getAll*` functions will return an empty `*Collection` in this case, in order to avoid the caller having to check for nil. Suggest same here.
```suggestion
	if pool.Spec.Inventory == nil {
		return &cdcCollection{}, nil
	}
```